applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.cpp

```diff
@@ -389,4 +396,51 @@ void ElasticIsotropic3D::CalculateCauchyGreenStrain(
     noalias(rStrainVector) = MathUtils<double>::StrainTensorToVector(E_tensor);
 }
 
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ElasticIsotropic3D::AddThermalEffect(
+    ConstitutiveLaw::Parameters& rValues,
+    Vector& rStrainVector
+    )
+{
+    const auto &r_properties = rValues.GetMaterialProperties();
+    const auto &r_geometry = rValues.GetElementGeometry();
+
+    if (r_properties.Has(THERMAL_EXPANSION_COEFFICIENT)) {
+        const double alpha = r_properties[THERMAL_EXPANSION_COEFFICIENT];
+        if (r_geometry[0].SolutionStepsDataHas(TEMPERATURE)) {
+            const double gauss_point_temperature = ConstitutiveLawUtilities<VoigtSize>::
+                GetNodalVariableOnIntegrationPoint(TEMPERATURE, rValues);
+            const double gauss_point_ref_temperature = ConstitutiveLawUtilities<VoigtSize>::
+                GetNodalVariableOnIntegrationPoint(REFERENCE_TEMPERATURE, rValues);
+            Vector thermal_strain_vector(VoigtSize);
+            this->ComputeThermalStrainVector(gauss_point_temperature, gauss_point_ref_temperature, 
+                alpha, thermal_strain_vector);
+            rStrainVector -= thermal_strain_vector;
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ElasticIsotropic3D::ComputeThermalStrainVector(
+    const double Temperature,
+    const double ReferenceTemperature,
+    const double ThermalExpansionCoeff,
+    Vector& rThermalStrainVector
+)
+{
+    if (rThermalStrainVector.size() != VoigtSize)
+        rThermalStrainVector.resize(VoigtSize);
+    rThermalStrainVector[0] = 1.0;
```

Comment:
i would add it here instead of multiplying the whole vector

Reply:
done, thank you